core/url.go

```diff
@@ -35,3 +41,59 @@ func JoinURLPaths(parts ...string) string {
 	}
 	return result
 }
+
+// ParsePublicURL parses the given input string as URL and asserts that
+// it has a scheme,
+// it is not an IP address, and
+// it is not a reserved address or TLD as described in RFC2606 or https://www.ietf.org/archive/id/draft-chapin-rfc2606bis-00.html.
+func ParsePublicURL(input string) (*url.URL, error) {
```

Comment:
can't this just be a private func in the auth module? Not really any reason to put it in core, since no other modules use it?